qualtran/bloqs/data_loading/one_hot_encoding_test.py

```diff
@@ -0,0 +1,57 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import attrs
+import cirq
+import pytest
+from attr import field
+from numpy._typing import NDArray
+
+from qualtran import GateWithRegisters, QUInt, Signature
+from qualtran.bloqs.data_loading.one_hot_encoding import OneHotEncoding
+from qualtran.cirq_interop.bit_tools import iter_bits
+from qualtran.cirq_interop.testing import assert_circuit_inp_out_cirqsim
+
+
+@attrs.frozen
+class OneHotEncodingTest(GateWithRegisters):
+    integer: int = field()
+    size: int = field()
+
+    @property
+    def signature(self) -> 'Signature':
+        return Signature.build_from_dtypes(a=QUInt(self.size), b=QUInt(2**self.size))
+
```

Comment:
I think it would be worthwhile to add on_classical_vals and add_my_tensors methods

Reply:
The action of one-hot encoding when the second register isn't in the zero state is not well-defined so the on_classical_vals and  add_my_tensors will be kind of unnatural. Is there a way to define the action only a subset of the basis states?

Reply:
- the classical simulation will always operate on computational basis states, so that should be appropriate here
 - I guess if the "target" isn't in the zero state, you'll get random things swapped around
     - one option is to change the output to a RIGHT register and allocate it inside the bloq. 
     - the other would be to raise an error in the classical simulation if the second register isn't in the zero state; but really -- this sounds like your abstraction is wrong and it should be a RIGHT register

Reply:
Ok this is what I was thinking. Is add_my_tensors still a thing when the bloq isn't unitary? (as it wouldn't be if I made a right register.)

Reply:
Just added on_classical_vals, should we also add add_my_tensors (also don't know if it applies given the non-unitary nature of this bloq now) @fdmalone

Reply:
nevermind added